COMMIT_GUIDELINES.md

```diff
@@ -0,0 +1,149 @@
+# General philosophy
+
+- **Each change goes into its own commit**. If you want to summarize what you did with this commit (in the commit message)
+and you start using the word "and", you probably want to split the commit up into 2 or more individual commits.
+
+- **Each commit should compile** (try to make each commit self-contained so that it is possible to compile the code at
+each commit)
+
+- **Take your time** when composing the commit message. In order to have a good git-history the commit messages are essential.
+Also if you put effort into the commit-message, you'll save work when creating a PullRequest as the description is already available.
+
+
+# Commit message
+
+Commit messages must follow the scheme
```

Comment:
```suggestion
Commit messages must follow the scheme:
```

Reply:
For a colon we'd also have to change `the` to `this`

Reply:
Done